app/models/story.rb

```diff
@@ -70,11 +76,10 @@ def set_last_update_time!(time=nil)
   end
 
   def self.for_user(user)
-    if user.nil? or user.sfw_filter
-      where("NOT adult").joins("LEFT OUTER JOIN watchlists ON watchlists.id = stories.watchlist_id").where("watchlists.id IS NULL OR watchlists.private = 'f'")
-    else
-      joins("LEFT OUTER JOIN watchlists ON watchlists.id = stories.watchlist_id").where("watchlists.id IS NULL OR watchlists.private = 'f'")
-    end
+    query = eager_load(:library_entry)
+            .where("library_entries.id IS NULL OR library_entries.private = 'f'")
```

Comment:
damn, I was pretty sure that there was a method to do this already… Unfortunately, we'll need to wait til Rails 5 for it :/